dev/com.ibm.ws.microprofile.metrics.2.3/src/com/ibm/ws/microprofile/metrics23/writer/PrometheusMetricWriter23.java

```diff
@@ -66,7 +66,10 @@ protected void writeMetricMapAsPrometheus(StringBuilder builder, String registry
             double conversionFactor = conversionAppendEntry.getValue();
             String appendUnit = conversionAppendEntry.getKey();
 
-            if (metricMetadata.getTypeRaw().equals(MetricType.COUNTER)) {
+            //If current metadata that we are parsing does not have a matching metric... skip
+            if (currentMetricMap.isEmpty()) {
```

Comment:
Should the empty check on crrentMetricMap move to line 56. If it is empty, you don't need to compute description, unit, conversionFactory etc.